bluemira/equilibria/coils/_coil.py

```diff
@@ -0,0 +1,521 @@
+        self._quad_dx = np.array([self.dx])
+        self._quad_dz = np.array([self.dz])
+        self._quad_weighting = np.ones_like(self._quad_x)
+        self._einsum_str = "...j, ...j -> ..."
+
+        if self.discretisation < 1:
+            # How fancy do we want the mesh or just smaller rectangles?
+            self._rectangular_discretisation()
+
+    def _validate_size(self):
+        dx_spec = is_num(self.dx)
+        dz_spec = is_num(self.dz)
+        dxdz_spec = dx_spec and dz_spec
+
+        if (dx_spec ^ dz_spec) and not dxdz_spec:
+            # Check that we don't have dx = None and dz = float or vice versa
+            raise EquilibriaError("Must specify either dx and dz or neither.")
+        if dxdz_spec:
+            # If dx and dz are specified, we presume the coil size should
+            # remain fixed
+            if not self._flag_sizefix:
+                self._flag_sizefix = True
+
+            self._set_coil_attributes()
+            self._discretise()
+        else:
+            if not is_num(self.j_max):
+                # Check there is a viable way to size the coil
+                raise EquilibriaError("Must specify either dx and dz or j_max.")
+
+            if self._flag_sizefix:
+                self._flag_sizefix = False
+
+    def _set_coil_attributes(self):
+        self._current_radius = 0.5 * np.hypot(self.dx, self.dz)
+        self._x_boundary, self._z_boundary = self._make_boundary(
+            self.x, self.z, self.dx, self.dz
+        )
+
+    def _rectangular_discretisation(self):
+        """
+        Discretise a coil into smaller rectangles based on fraction of
+        coil dimensions
+
+        Parameters
+        ----------
+        d_coil: float
+            Target discretisation fraction
+
+        Returns
+        -------
+        weighting: np.ndarray
+
+        """
+        nx = np.maximum(1, np.ceil(self.dx * 2 / self.discretisation))
+        nz = np.maximum(1, np.ceil(self.dz * 2 / self.discretisation))
+
+        if not nx * nz == 1:
+            sc_dx, sc_dz = self.dx / nx, self.dz / nz
+
+            # Calculate sub-coil centroids
+            x_sc = (self.x - self.dx) + sc_dx * np.arange(1, 2 * nx, 2)
+            z_sc = (self.z - self.dz) + sc_dz * np.arange(1, 2 * nz, 2)
+            x_sc, z_sc = np.meshgrid(x_sc, z_sc)
+
+            self._quad_x = x_sc.flatten()
+            self._quad_z = z_sc.flatten()
+            self._quad_dx = np.ones(x_sc.size) * sc_dx
+            self._quad_dz = np.ones(x_sc.size) * sc_dz
+
+            self._quad_weighting = np.ones(x_sc.size) / x_sc.size
+
+    def fix_size(self):
+        """
+        Fixes the size of all coils
+        """
+        self._flag_sizefix = True
+
+    def resize(self, current: Optional[float] = None):
+        """Resize coil given a current"""
+        if not self._flag_sizefix:
+            # Adjust the size of the coil
+            self.dx, self.dz = self._make_size(current)
+            self._set_coil_attributes()
```

Comment:
Fair point. I will add a message to the initialisation if dx and dz are set that way at least there is one info message. The init dx and dz just stops the recalculation of the coil size if the current changes.

It almost feels like a debug message to me though because if you set the coil size its a definitive action that has been taken.